catroid/src/main/java/org/catrobat/catroid/cloudmessaging/CloudMessaging.java

```diff
@@ -54,6 +69,50 @@ public void showNotification(CloudMessage cloudMessage) {
 		}
 	}
 
+	public static boolean initialize(Context context) {
+		if (isCloudNotificationsEnabled(context)) {
+			subscribeToFirebaseServices(context);
+			return true;
+		}
+		unsubscribeFromFirebaseServices(context);
+		return false;
+	}
+
+	@VisibleForTesting
+	public static void setIsCloudNotificationsEnabled(boolean isEnabled) {
+		isCloudNotificationsEnabled = isEnabled;
+	}
+
+	private static void subscribeToFirebaseServices(Context context) {
+		FirebaseMessaging.getInstance().subscribeToTopic(DEFAULT_TOPIC);
+		FirebaseMessaging.getInstance().subscribeToTopic(getLanguageCode(context));
+		FirebaseMessaging.getInstance().subscribeToTopic(getCountryCode(context));
+	}
+
+	private static void unsubscribeFromFirebaseServices(Context context) {
+		FirebaseMessaging.getInstance().unsubscribeFromTopic(DEFAULT_TOPIC);
+		FirebaseMessaging.getInstance().unsubscribeFromTopic(getLanguageCode(context));
+		FirebaseMessaging.getInstance().unsubscribeFromTopic(getCountryCode(context));
+	}
+
+	private static boolean isCloudNotificationsEnabled(Context context) {
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		return sharedPreferences.getBoolean(SettingsActivity.SETTINGS_CLOUD_NOTIFICATIONS, false) && isCloudNotificationsEnabled;
+	}
+
+	private static String getLanguageCode(Context context) {
+		String languageCode = context.getResources().getConfiguration().locale.getCountry();
+		Log.v(TAG, "Language Code : " + languageCode);
+		return languageCode;
+	}
+
+	private static String getCountryCode(Context context) {
+		TelephonyManager telephonyManager = (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE);
+		String countryCode = telephonyManager.getNetworkCountryIso();
+		Log.v(TAG, "Country Code : " + countryCode);
+		return countryCode;
+	}
+
```

Comment:
CloudMessaging handles
* Building and showing notifications
* Un-/subscribing to topics

IMO this can easily be split into two classes.